### PR TITLE
repo reorg: move tcp_proxy code code back

### PR DESCRIPTION
**Pull request by mattklein123**
It was a mistake to move it in the first place as it must be included
in the core server. This completes the repo reorg project.

Fixes https://github.com/envoyproxy/envoy/issues/3281
Fixes https://github.com/envoyproxy/envoy/issues/2910

*Risk Level*: Low
*Testing*: Existing tests
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@ggreenway

File: source/extensions/filters/network/tcp_proxy/config.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 namespace Envoy {
 namespace Extensions {
 namespace NetworkFilters {
-namespace TcpProxy {
+namespace TcpProxyFilter {
```
Comment (ggreenway) [inline source/extensions/filters/network/tcp_proxy/config.h]: Why this change? I don't think any of the other network filters are named like this.

Comment (mattklein123) [reply source/extensions/filters/network/tcp_proxy/config.h]: It was because of namespace conflicts. With the other changes I can change it back.
